src/02/fp-math.asm

```diff
@@ -268,21 +267,25 @@ _:
 ;; Inputs:
 ;;  IX: Pointer to floating point number
 ;;  HL: Pointer to destination buffer
-;;  A: Flags
+;;  A: Flags / digits
 ;; Notes:
-;;  The destination buffer must be large enough to store the result.
+;;  The destination buffer must be at least 20 characters in length.
+;;
+;;  The most significant nibble of A should be set to flags. FP_STR_*
+;;  macros in kernel.inc can be ORed and stored in A for convenience.
 ;;
-;;  The flag byte consists of the following bits:
-;;  * 7: Set to use comma as decimal point and period as place value separator
-;;  * 6: Set to display place value separators
-;;  * 5-4: Display mode. 00 = Normal and 01 = Scientific. Modes 10 and 11
-;;         are currently undefined.
-;;  * 3-0: Number of fixed point digits to display after the decimal (0-9),
-;;         or 0xF for float mode. Works like TI-OS's Fix/Float mode.
+;;  The least significant nibble of A is the number of digits to display
+;;  after the decimal: 0-9 for that many digits, or 0xF for as many are
+;;  non-zero.
 ;;
-;; TODO:
-;;  * Rounding last digit - buggy, currently abandoned
-;;  * Never show exponent if significand is 0 - not started
+;;  Examples:
+;;  All nonzero decimals in scientific notation, digits grouped with ',':
```

Comment:
This example doesn't make much sense because scientific notation will never have thousands separators. Might make more sense to change this to FP_STR_INV_PUNC | FP_DISP_SCIENTIFIC | 0xF, and then make the next one FP_GROUP_DIGITS | 5